joern-cli/frontends/php2cpg/src/test/scala/io/joern/php2cpg/passes/PhpSymbolSummaryPassTest.scala

```diff
@@ -0,0 +1,144 @@
+        name shouldBe NamespaceTraversal.globalNamespaceName
+        fooMethod.name shouldBe "foo"
+      }
+    )
+  }
+
+  "pre-parsing a file with a top-level class with a nested function should provide a summary of that class" in {
+    assertAgainstTempFile(
+      """<?php
+        |class Foo {
+        |  final public function foo(int $x): int {
+        |    return 0;
+        |  }
+        |}
+        |""".stripMargin :: Nil,
+      {
+        case PhpNamespace(
+              name,
+              PhpClass(className, (constr: PhpFunction) :: (fooMethod: PhpFunction) :: Nil) :: Nil
+            ) :: Nil =>
+          name shouldBe NamespaceTraversal.globalNamespaceName
+          className shouldBe "Foo"
+          constr.name shouldBe Domain.ConstructorMethodName
+          fooMethod.name shouldBe "foo"
+      }
+    )
+  }
+
+  "pre-parsing a file with a top-level class with a nested constant should provide a summary of that class" in {
+    assertAgainstTempFile(
+      """<?php
+        |class Foo {
+        |  const B = "B";
+        |}
+        |""".stripMargin :: Nil,
+      {
+        case PhpNamespace(
+              name,
+              PhpClass(className, (constr: PhpFunction) :: (bConst: PhpMember) :: Nil) :: Nil
+            ) :: Nil =>
+          name shouldBe NamespaceTraversal.globalNamespaceName
+          className shouldBe "Foo"
+          constr.name shouldBe Domain.ConstructorMethodName
+          bConst.name shouldBe "B"
+      }
+    )
+  }
+
+  "pre-parsing a file with a nested namespace declaration should be summarized accordingly" in {
+    assertAgainstTempFile(
+      """<?php
+        |namespace Foo\Bar;
+        |class Baz {}
+        |""".stripMargin :: Nil,
+      {
+        case PhpNamespace(
+              globalName,
+              PhpNamespace(fooName, PhpNamespace(barName, PhpClass(bazName, _) :: Nil) :: Nil) :: Nil
+            ) :: Nil =>
+          globalName shouldBe NamespaceTraversal.globalNamespaceName
+          fooName shouldBe "Foo"
+          barName shouldBe "Bar"
+          bazName shouldBe "Baz"
+      }
+    )
+  }
+
+  "pre-parsing a file with a duplicate namespace declaration should be summarized and deduplicated accordingly" in {
+    assertAgainstTempFile(
+      """<?php
+        |namespace Foo\Bar;
+        |class Baz {}
+        |""".stripMargin ::
+        """<?php
+        |namespace Foo\Bar;
+        |class Faz {}
+        |""".stripMargin :: Nil,
+      {
+        case PhpNamespace(
+              globalName,
+              PhpNamespace(
+                fooName,
+                PhpNamespace(barName, PhpClass(bazName, _) :: PhpClass(fazName, _) :: Nil) :: Nil
+              ) :: Nil
+            ) :: Nil =>
+          globalName shouldBe NamespaceTraversal.globalNamespaceName
+          fooName shouldBe "Foo"
+          barName shouldBe "Bar"
+          bazName shouldBe "Baz"
+          fazName shouldBe "Faz"
+      }
+    )
+  }
+
```

Comment:
Missing a test about functions nested in namespace.
If that is possible we should also have a test of namespace in namespace.

Reply:
Added a couple of tests, and improved deduplication when separated namespaces are adjacent to one another